src/spatial/delaunay.c

```diff
@@ -0,0 +1,247 @@
+    }
+    // first are equal
+    if (A[1] < B[1]) {
+        return -1;
+    }
+    if (A[1] > B[1]) {
+        return  1;
+    }
+
+    // second are equal
+    return 0;
+}
+
+// Simplify an edge list in place
+// except is the vertex "at infinity", and should not be included in the edge list.
+static igraph_error_t simplify_edge_list(igraph_vector_int_t *in, igraph_integer_t except) {
+    igraph_integer_t size = igraph_vector_int_size(in);
+    if (size == 0) {
+        return IGRAPH_SUCCESS;
+    }
+    // reorder
+    igraph_integer_t temp;
+    for (igraph_integer_t i = 0; i < size; i += 2) {
+        if (VECTOR(*in)[i] > VECTOR(*in)[i + 1]) {
+            temp = VECTOR(*in)[i];
+            VECTOR(*in)[i] = VECTOR(*in)[i+1];
+            VECTOR(*in)[i+1] = temp;
+        }
+    }
+    // sort
+    igraph_qsort(VECTOR(*in), size / 2, 2 * sizeof(igraph_integer_t), &edge_comparator);
+    // nub and filter out point at infinity
+    igraph_integer_t last_added = 0;
+    for (igraph_integer_t i = 2; i < size; i += 2) {
+        if ((VECTOR(*in)[i] != VECTOR(*in)[2 * last_added] || VECTOR(*in)[i + 1] != VECTOR(*in)[2*last_added + 1]) && VECTOR(*in)[i+1] != except) {
+            last_added += 1;
+            VECTOR(*in)[2*last_added]      = VECTOR(*in)[i];
+            VECTOR(*in)[2*last_added + 1 ] = VECTOR(*in)[i+1];
+        }
+    }
+    IGRAPH_CHECK(igraph_vector_int_resize(in, 2*last_added+2));
+    return IGRAPH_SUCCESS;
+}
+
+
+// helper that can go on the finally stack to free qhull
+static void destroy_qhull(qhT *qh) {
+    int curlong, totlong;
+    qh->NOerrexit = True; /* no more setjmp */
+    qh_freeqhull(qh, !qh_ALL);
+    qh_memfreeshort(qh, &curlong, &totlong);
+}
+
+// Make a transposed copy with space for added point at infinity.
+static igraph_error_t copy_transpose(const igraph_matrix_t *in, igraph_matrix_t *out) {
+    igraph_integer_t rows = igraph_matrix_nrow(in), columns = igraph_matrix_ncol(in);
+
+    // add extra space for point at infinity
+    IGRAPH_CHECK(igraph_matrix_init(out, columns, rows+1));
+
+    for (igraph_integer_t row = 0; row < rows; row++) {
+        for (igraph_integer_t col = 0; col < columns; col++) {
+            MATRIX(*out, col, row) = MATRIX(*in, row, col);
+        }
+    }
+
+    return IGRAPH_SUCCESS;
+}
+
+
+igraph_error_t igraph_i_delaunay_edges(igraph_vector_int_t *edges, const igraph_matrix_t *points) {
+    int exitcode;
+    igraph_integer_t numpoints = igraph_matrix_nrow(points) ;
+    igraph_integer_t dim = igraph_matrix_ncol(points);
+    boolT ismalloc = False; // handle memory allocation of points explicitly
+    qhT qh_qh;
+    qhT *qh = &qh_qh;
+
+    igraph_matrix_t int_points;
+
+    if (numpoints > INT_MAX) {
+        IGRAPH_ERROR("Too many points for Qhull.", IGRAPH_EOVERFLOW);
+    }
+
+    // For internal use, it returns a complete graph.
+    // This is not nessecarily a delaunay graph, but for nondegenerate point sets it is.
+    if (numpoints <= dim) {
+
+    }
```

Comment:
@Zepeacedust It seems this is not yet complete. I'm marking the PR as a draft again. Let me know when it's ready to go.